scripts/generate_ci_matrix.py

```diff
@@ -24,24 +24,28 @@
 SCHEDULE_MATRICES: dict[str, list[dict[str, str]]] = matrix_config.get("schedules", {})
 
 
-def generate_matrix(schedule="", test_setup=""):
+def generate_matrix(schedule="", test_setup="", matrix_key="default"):
     """
-    Generate test matrix based on schedule or manual input.
+    Generate test matrix based on schedule, manual input, or matrix key.
 
     Args:
         schedule: GitHub cron schedule string (e.g., "1 0 * * 0" for weekly)
         test_setup: Manual test setup input (e.g., "ollama-vision")
+        matrix_key: Matrix configuration key from ci_matrix.json (e.g., "default", "stainless")
 
     Returns:
         Matrix configuration as JSON string
     """
-    # Weekly scheduled test matrices
+    # Weekly scheduled test matrices (highest priority)
     if schedule and schedule in SCHEDULE_MATRICES:
         matrix = SCHEDULE_MATRICES[schedule]
     # Manual input for specific setup
     elif test_setup == "ollama-vision":
         matrix = [{"suite": "vision", "setup": "ollama-vision"}]
-    # Default: use JSON-defined matrix
+    # Use specified matrix key from ci_matrix.json
+    elif matrix_key and matrix_key in matrix_config:
```

Comment:
you should error if the key is not in the config, otherwise you are going to just skip over into the `else` branch. silent error eating is not a good idea.

Reply:
changed